File: app-mvvm/src/main/java/com/xuie/imaginary/gui/nets/detail/NetsOneContract.java
package com.xuie.imaginary.gui.nets.detail;

import com.xuie.imaginary.BasePresenter;
import com.xuie.imaginary.BaseView;
import com.xuie.imaginary.data.NetsDetail;

/**
 * This specifies the contract between the view and the presenter.
 *
 * @author xuie
 */
public interface NetsOneContract {

    interface View extends BaseView {
        void refreshNewsOne(NetsDetail netsDetail);
    }

    interface Presenter extends BasePresenter {
        void getNewsOneRequest(String postId);
    }
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/gui/nets/detail/NetsOneContract.kt
package com.xuie.imaginaryandroid.gui.nets.detail

import com.xuie.imaginaryandroid.BasePresenter
import com.xuie.imaginaryandroid.BaseView
import com.xuie.imaginaryandroid.data.NetsDetail

/**
 * This specifies the contract between the view and the presenter.
 */
interface NetsOneContract {

    interface View : BaseView<Presenter> {
        fun refreshNewsOne(netsDetail: NetsDetail)
    }

    interface Presenter : BasePresenter {
        fun getNewsOneRequest(postId: String)
    }
}
